Source/PeleCAmr.cpp

```diff
@@ -0,0 +1,232 @@
+    for (int comp = 0; comp < desc_lst[typ].nComp(); comp++) {
+      if (
+        amrex::Amr::isStatePlotVar(desc_lst[typ].name(comp)) &&
+        desc_lst[typ].getType() == amrex::IndexType::TheCellType()) {
+        plot_var_map.push_back(std::pair<int, int>(typ, comp));
+      }
+    }
+  }
+
+  int num_derive = 0;
+  auto& derive_lst = amrex::AmrLevel::get_derive_lst();
+  std::list<std::string> derive_names;
+  const std::list<amrex::DeriveRec>& dlist = derive_lst.dlist();
+  if (regular) {
+    for (const auto& it : dlist) {
+      if (amrex::Amr::isDerivePlotVar(it.name())) {
+        derive_names.push_back(it.name());
+        num_derive += it.numDerive();
+      }
+    }
+  }
+
+  const auto n_data_items =
+    plot_var_map.size() + num_derive +
+    static_cast<int>(amrex::EB2::TopIndexSpaceIfPresent() != nullptr);
+
+  const int nGrow = 0;
+  const amrex::Real cur_time =
+    (amr_level[0]->get_state_data(State_Type)).curTime();
+
+  const int nlevels = finestLevel() + 1;
+  amrex::Vector<std::unique_ptr<amrex::MultiFab>> plotMFs(nlevels);
+  amrex::Vector<int> istep(nlevels);
+  for (int lev = 0; lev < nlevels; ++lev) {
+
+    istep[lev] = levelSteps(lev);
+
+    plotMFs[lev] = std::make_unique<amrex::MultiFab>(
+      boxArray(lev), DistributionMap(lev), n_data_items, nGrow, amrex::MFInfo(),
+      amr_level[lev]->Factory());
+
+    // Cull data from state variables -- use no ghost cells.
+    int cnt = 0;
+    for (int i = 0; i < plot_var_map.size(); i++) {
+      int typ = plot_var_map[i].first;
+      int comp = plot_var_map[i].second;
+      amrex::MultiFab& this_dat = amr_level[lev]->get_new_data(typ);
+      amrex::MultiFab::Copy(*plotMFs[lev], this_dat, comp, cnt, 1, nGrow);
+      cnt++;
+    }
+
+    // Cull data from derived variables.
+    if ((!derive_names.empty())) {
+      for (const auto& derive_name : derive_names) {
+        const amrex::DeriveRec* rec = derive_lst.get(derive_name);
+        int ncomp = rec->numDerive();
+
+        auto derive_dat = amr_level[lev]->derive(derive_name, cur_time, nGrow);
+        amrex::MultiFab::Copy(*plotMFs[lev], *derive_dat, 0, cnt, ncomp, nGrow);
+        cnt += ncomp;
+      }
+    }
+
+    if (amrex::EB2::TopIndexSpaceIfPresent() != nullptr) {
+      const auto& ebfactory = dynamic_cast<amrex::EBFArrayBoxFactory const&>(
+        amr_level[lev]->Factory());
+      amrex::MultiFab::Copy(
+        *plotMFs[lev], ebfactory.getVolFrac(), 0, cnt, 1, nGrow);
+      cnt++;
+    }
+  }
+
+  amrex::Vector<std::string> plt_var_names;
+  for (int i = 0; i < plot_var_map.size(); i++) {
+    int typ = plot_var_map[i].first;
+    int comp = plot_var_map[i].second;
+    plt_var_names.push_back(desc_lst[typ].name(comp));
+  }
+
+  for (const auto& derive_name : derive_names) {
+    const amrex::DeriveRec* rec = derive_lst.get(derive_name);
+    for (int i = 0; i < rec->numDerive(); i++) {
+      plt_var_names.push_back(rec->variableName(i));
+    }
+  }
+
```

Comment:
And then here we need to `plt_var_names.push_back("vfrac")` under the same conditions as above. We should probably add a check that `plt_var_names.size() == n_data_items` after these `push_back`s to ensure everything was done right.